base_layer/core/src/base_node/state_machine_service/states/listening.rs

```diff
@@ -86,6 +91,11 @@ impl Listening {
                         let sync_peers = select_sync_peers(local_tip_height, &best_metadata, &peer_metadata_list);
 
                         let sync_mode = determine_sync_mode(&local, best_metadata, sync_peers);
+                        if sync_mode == SyncStatus::UpToDate &&
```

Comment:
I don't think this step is necessary. The base node being in `Listening` state is IMHO sufficient to determine that the base node is synced. If `sync_mode.is_lagging()` is true in the next statement, the state will transition out of `Listening`

Reply:
Unfortunately it is because the State Machine starts in StartUp transitions to Listening and then only transitions to Lagging after a full round of ChainMetadata pings has been received from the neighbourhood peers. This means that for a short but not insignificant time the State Machine is in Listening state but not synced.

Reply:
Agreed with @philipr-za